### ISSUE TITLE
Retrieving default `arbitrum_one` rpc url fails

### ISSUE THREAD

**Issue by ramenforbreakfast** [foundry-rs/foundry#7491]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (88e09f6 2024-03-25T00:21:20.717965000Z)

### Operating System

macOS (Intel)

### Describe the bug

I have been using the `forge-safe` library to write `forge` scripts that can build multi-sig proposals for my team to review and execute.

For the past month, I haven't had any problems using it against the Arbitrum One network. However, recently, whenever I try to run the script against Arbitrum One, I receive a "invalid rpc url: arbitrum_one" error.

After a bit of debugging, I have pinpointed the issue to the `getChain(string memory chainAlias)` call that `forge-safe`'s `BatchScript.sol` calls. `forge-safe` requires you to specify the chain alias of the network you want to submit the safe proposal for in the `CHAIN` environment variable. It then calls `getChain(<CHAIN env var>)` to retrieve the network's information.

Even though I have set the `CHAIN` environment variable to `arbitrum_one`, it fails to retrieve the rpc url. According to `forge-std` `StdChains.sol`, `arbitrum_one` is a default `chainAlias` that should be initialized properly.

I even tried to also set an env entry for `ARBITRUM_ONE_RPC_URL='https://arb1.arbitrum.io/rpc'` as a fallback and it failed to recognize it.

I also tried to isolate whether it was an issue with `forge-safe`'s own `forge-std` dependency, by writing a barebones script to `console.log(vm.rpcUrl("arbitrum_one"))` and that failed in the same manner as well.

What seems to solve it for now, is if I define it again in my `foundry.toml`
```
[rpc_endpoints]
arbitrum_one = 'https://arb1.arbitrum.io/rpc'
```

For further context, here is the `getChain` callstack, before and after I defined the above entry in my `foundry.toml`.
BEFORE
```
  [2513225] DeallocateAuctions::submitDeallocateProposal(0, 1)
    ├─ [0] VM::envString("CHAIN") [staticcall]
    │   └─ ← [Return] <env var value>
    ├─ [0] VM::toString(31337 [3.133e4]) [staticcall]
    │   └─ ← [Return] "31337"
    ├─ [0] VM::toString(1) [staticcall]
    │   └─ ← [Return] "1"
    ├─ [0] VM::toString(5) [staticcall]
    │   └─ ← [Return] "5"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(10) [staticcall]
    │   └─ ← [Return] "10"
    ├─ [0] VM::toString(420) [staticcall]
    │   └─ ← [Return] "420"
    ├─ [0] VM::toString(42161 [4.216e4]) [staticcall]
    │   └─ ← [Return] "42161"
    ├─ [0] VM::toString(421613 [4.216e5]) [staticcall]
    │   └─ ← [Return] "421613"
    ├─ [0] VM::toString(42170 [4.217e4]) [staticcall]
    │   └─ ← [Return] "42170"
    ├─ [0] VM::toString(137) [staticcall]
    │   └─ ← [Return] "137"
    ├─ [0] VM::toString(80001 [8e4]) [staticcall]
    │   └─ ← [Return] "80001"
    ├─ [0] VM::toString(43114 [4.311e4]) [staticcall]
    │   └─ ← [Return] "43114"
    ├─ [0] VM::toString(43113 [4.311e4]) [staticcall]
    │   └─ ← [Return] "43113"
    ├─ [0] VM::toString(56) [staticcall]
    │   └─ ← [Return] "56"
    ├─ [0] VM::toString(97) [staticcall]
    │   └─ ← [Return] "97"
    ├─ [0] VM::toString(100) [staticcall]
    │   └─ ← [Return] "100"
    ├─ [0] VM::rpcUrl("arbitrum_one") [staticcall]
    │   └─ ← [Revert] invalid rpc url: arbitrum_one
    ├─ [0] VM::envOr("ARBITRUM_ONE_RPC_URL", "https://arb1.arbitrum.io/rpc")
    │   └─ ← [Return] <env var value>
    └─ ← [Revert] invalid rpc url: arbitrum_one
```

AFTER
```
[2671796] DeallocateAuctions::submitDeallocateProposal(0, 1)
    ├─ [0] VM::envString("CHAIN") [staticcall]
    │   └─ ← [Return] <env var value>
    ├─ [0] VM::toString(31337 [3.133e4]) [staticcall]
    │   └─ ← [Return] "31337"
    ├─ [0] VM::toString(1) [staticcall]
    │   └─ ← [Return] "1"
    ├─ [0] VM::toString(5) [staticcall]
    │   └─ ← [Return] "5"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(10) [staticcall]
    │   └─ ← [Return] "10"
    ├─ [0] VM::toString(420) [staticcall]
    │   └─ ← [Return] "420"
    ├─ [0] VM::toString(42161 [4.216e4]) [staticcall]
    │   └─ ← [Return] "42161"
    ├─ [0] VM::toString(421613 [4.216e5]) [staticcall]
    │   └─ ← [Return] "421613"
    ├─ [0] VM::toString(42170 [4.217e4]) [staticcall]
    │   └─ ← [Return] "42170"
    ├─ [0] VM::toString(137) [staticcall]
    │   └─ ← [Return] "137"
    ├─ [0] VM::toString(80001 [8e4]) [staticcall]
    │   └─ ← [Return] "80001"
    ├─ [0] VM::toString(43114 [4.311e4]) [staticcall]
    │   └─ ← [Return] "43114"
    ├─ [0] VM::toString(43113 [4.311e4]) [staticcall]
    │   └─ ← [Return] "43113"
    ├─ [0] VM::toString(56) [staticcall]
    │   └─ ← [Return] "56"
    ├─ [0] VM::toString(97) [staticcall]
    │   └─ ← [Return] "97"
    ├─ [0] VM::toString(100) [staticcall]
    │   └─ ← [Return] "100"
    ├─ [0] VM::rpcUrl("arbitrum_one") [staticcall]
    │   └─ ← [Return] "https://arb1.arbitrum.io/rpc"
    ├─ [0] VM::envString("WALLET_TYPE") [staticcall]
```

 ... `forge-safe` proceeds to retrieve other info now that the chain retrieval suceeded.

**Comment by mattsse** [comment]
do you have a repro for this? I'm unable to reproduce this via ` console.log(vm.rpcUrl("arbitrum_one"));`

```
[PASS] test_Increment() (gas: 6715)
Logs:
  https://arb1.arbitrum.io/rpc
```

**Comment by ramenforbreakfast** [comment]
Ah sorry, been debugging further, I realized that I think the problem is that my `forge-std` is out of date (1.5.2) and my `forge update lib/forge-std` doesn't seem to be pulling down the latest version. I'll close this issue since that's an entirely different problem!

**Comment by ramenforbreakfast** [comment]
So I managed to update my `forge-std` to 1.8.1 (had to sync the git submodule, had the wrong remote path). I am still encountering this problem.

Further debugging has me thinking this might be related to problem with a prior cached rpc entry?

When I attempted to call `forge cache clean` I got this issue

```
error: invalid value 'arbitrum_one' for '[CHAINS]...'
  [possible values: all, mainnet, morden, ropsten, rinkeby, goerli, kovan, holesky, sepolia, optimism, optimism-kovan, optimism-goerli, optimism-sepolia, arbitrum, arbitrum-testnet, arbitrum-goerli, arbitrum-sepolia, arbitrum-nova, cronos, cronos-testnet, rsk, bsc, bsc-testnet, poa, sokol, scroll, scroll-alpha-testnet, scroll-sepolia, metis, xdai, polygon, mumbai, amoy, polygon-zkevm, polygon-zkevm-testnet, fantom, fantom-testnet, moonbeam, moonbeam-dev, moonriver, moonbase, dev, anvil-hardhat, evmos, evmos-testnet, chiado, oasis, emerald, emerald-testnet, filecoin-mainnet, filecoin-calibration-testnet, avalanche, fuji, celo, celo-alfajores, celo-baklava, aurora, aurora-testnet, canto, canto-testnet, boba, base, base-goerli, base-sepolia, syndr-sepolia, fraxtal, fraxtal-testnet, blast, blast-sepolia, linea, linea-goerli, zksync, zksync-testnet, mantle, mantle-testnet, viction, zora, zora-goerli, zora-sepolia, pgn, pgn-sepolia, mode, mode-sepolia, elastos, kakarot-sepolia]

  tip: a similar value exists: 'arbitrum'

For more information, try '--help'.
```

**Comment by mattsse** [comment]
is this now a CLI parser error?

unclear how to reproduce this

**Comment by ramenforbreakfast** [comment]
> is this now a CLI parser error?
> 
> unclear how to reproduce this

Well this is just another problem Im encountering that might be related to the originally posted issue. I was just attempting to clear my cache because somehow it recognized my `ARBITRUM_ONE_RPC_URL` env entry and has cached that to attribute `arbitrum_one` to it, preventing me from being able to reproduce the original issue.

However, it seems that it won't allow me to do so because `arbitrum_one` is not a valid `CHAIN`? I assume `forge cache clean` loops through every chain within some registry and attempts to clean the cache associated with that chain?

**Comment by mattsse** [comment]
could you try using just "arbitrum"?

**Comment by ramenforbreakfast** [comment]
Ok sorry, I think the problem has been solved for now. I think I got confused between `getChain` and `vm.rpcUrl`. If i'm not mistaken, when you attempted to reproduce via the testcase with `vm.rpcUrl`, that is actually attempting to retrieve the rpc url from your `foundry.toml`?

I'm assuming this based on the following (no chains were printed since I don't have any configured in my foundry.toml):
```
function test_Rpc() public {
  string[2][] memory chains = vm.rpcUrls();
  console.log("Printing Chains");
  for (uint256 i = 0; i < chains.length; i++) {
    string memory chain = chains[i][0];
    string memory rpc = chains[i][1];
    console.log("chain: %s, rpc: %s", chain, rpc);
  }
  assertEq(false, true);
}

Logs:
  Printing Chains

Traces:
  [6990] Repro_Fuzz::test_Rpc()
```

`getChain` will initialize default rpc entries first and then attempt to override any of them using `vm.rpcUrl` (which would be any entries you might have configured yourself).

`getChain` seems to be working now with `arbitrum_one` that I've updated to 1.8.1 so we can close this issue!
